src/redux/book/book.js

```diff
@@ -0,0 +1,62 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { postData, getData, removeData } from '../../components/services/service';
+
+const GET_BOOK = 'BookStore/book/GET_BOOK';
+const REMOVE_BOOK = 'BookStore/book/REMOVE_BOOK';
+const POST_BOOK = 'BookStore/book/POST_BOOK';
+
+const initialState = {
+  books: [],
+};
+
+const bookReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case `${POST_BOOK}/fulfilled`:
+      return {
+        ...state,
+        books: [...state.books, action.payload],
+
+      };
+
+    case `${GET_BOOK}/fulfilled`:
+      return {
+        ...state,
+        books: action.payload,
+      };
+
+    case `${REMOVE_BOOK}/fulfilled`:
+      return {
+        ...state,
+        books: state.books.filter((book) => book.itemId !== action.payload),
+      };
+
+    default:
+      return state;
+  }
+};
+
+export const postBook = createAsyncThunk(POST_BOOK, async (book) => {
+  const bookObj = {
+    ...book,
+    itemId: Date.now(),
+  };
+  await postData(bookObj);
```

Comment:
- [ ] You did a good job refactoring your action creators to use async thunks! 💯 However, I'm getting an error from the server whenever I try to create a new book and is happening because the API is expecting a property named `"item_id"` instead of the one you're passing which is `"itemId"` please consider fixing this! 🤓 🛠️ 

![image](https://user-images.githubusercontent.com/77069011/211054100-4a3e7859-3a3e-4fe3-9bd8-2c553dda1ca7.png)